### ISSUE TITLE
Animated drawers

### ISSUE THREAD

**Issue by tacotexmex** [minetest-mods/drawers#6]
Since we have the sound, how about animate the drawers too? It doesn't even have to slide, it may just pop out and back in again.

**Comment by LNJ2** [comment]
That would be a nice feature, but nothing I want to do soon.
(Also if you just add/remove a stack, how should the drawer be animated, a short open/close or what?)

**Comment by tacotexmex** [comment]
> a short open/close
Yes.

Or perhaps it's not viable, but then the sounds should change as well. ¯\_(ツ)_/¯

**Comment by LNJ2** [comment]
> I realize that this with physical attributes such as motion and sounds would make the most sense if item(s) dropped in-world in front of the drawer rather than adding them to the inventory directly. Which is a bigger project, I know. :D

I'm sure nobody wants to pick up the items from the ground... --> too realistic games are no fun

**Comment by tacotexmex** [comment]
I do! I always play with item drops enabled combined with your key-enabled item pickup code from TNG. It's excellent for multiplayer cooperation.

**Comment by LNJ2** [comment]
Yeah with an item drop mod that would be ok, but not with the default MTG.

**Comment by eshattow** [comment]
How about a compromise where the drawer handles change colours based on access (like computer HDD storage device LED access light from 1990's :-)
Another idea heard was to show when drawers are full with a graphic, but really this is an animation. I don't know what colours or animation to suggest but it can be a very simple change of color where the drawer handles are as if it is an indicator light with a few seconds timeout.

Maybe:
Default LED handle state not full, no drawer controller connected = As-is how the handles are now (or "indicator light off")
Default LED handle state not full with drawer controller = Blue continuous
Default LED handle state full = Red

When item stored recently but drawer not full then pulse Green on for a few seconds.
When item retrieved recently then pulse Yellow on for a few seconds.
If both Green and Yellow condition then color is Orange while those conditions overlap.
When item failed to store because drawer is full then blinking Red for a few seconds (this could be Red/off, Red/Green, Red/Yellow, Red/Orange, Red/Blue for certain corner cases). Red means full. Blinking Red means some kind of activity failed.
